internal/venice-common/src/main/java/com/linkedin/venice/meta/ReadOnlyStoreView.java

```diff
@@ -0,0 +1,143 @@
+package com.linkedin.venice.meta;
+
+import com.linkedin.venice.exceptions.StoreVersionNotFoundException;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.views.MaterializedView;
+import com.linkedin.venice.views.VeniceView;
+import com.linkedin.venice.views.ViewUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Properties;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+
+public class ReadOnlyStoreView extends ReadOnlyStore {
+  private final Map<Integer, ReadOnlyVersion> viewVersionMap;
+
+  public ReadOnlyStoreView(Store delegate, Map<Integer, ReadOnlyVersion> viewVersionMap) {
+    super(delegate);
+    this.viewVersionMap = viewVersionMap;
```

Comment:
Can we move the logic from `NativeMetadataRepositoryViewAdapter#getStore` to this function?
Essentially `store` contains everything we need and we can do all the decoration work here.